src/frontend/src/features/forms/components/message-form/index.tsx

```diff
@@ -381,16 +413,24 @@ export const MessageForm = ({
             return;
         }
 
-        const draft = await saveDraft(data);
+        // Ensure any pending auto-save operations complete before sending
+        // This prevents race conditions where editor changes haven't been saved yet
+        await new Promise(resolve => setTimeout(resolve, 350)); // Wait for debounce + buffer
```

Comment:
This is not bullet proof. Instead if `isSaving` is `true` we could just recall `handleSubmit` through a timeout WDYT ?

Reply:
```suggestion
        if (isSaving) {
            // If currently saving, wait and retry submit
            setTimeout(() => handleSubmit(data), 500);
            return;
        }
```
?